### PR TITLE
fix: 4722 compareAtPrice - convert from Float to Money

### LINKED ISSUE #4722: CatelogItem: compareAtPrice does not have currency symbol. No displayCompareAtPrice exists
## Issue Description

`displayCompareAtPrice` in the `pricing` object doesn't exist, so getting the `compareAtPrice` with the currency symbol as a string is not as easy as the `displayPrice`.

## Steps to Reproduce

1. Set the "compare at price" for a variant
2. Publish
3. Look in the DB or make a GQL query for that catalog item that contains the updated variant.
4. See only a `float` type for the "compare at price"

## Possible Solution

Add a `displayCompareAtPrice` with the proper currency symbol

### Versions

Node: 10.7.0
NPM: 6.2.0
Meteor Node: 8.11.4
Meteor NPM: 6.4.1
Reaction CLI: 0.29.0
Reaction: 2.0.0-rc.2
Reaction branch: master
Docker: 18.06.1-ce

### PR DESCRIPTION
**Pull request by mikemurray**
Resolves #4722   
Impact: **minor**  
Type: **bugfix**

## Issue

`compareAtPrice` is not of type `Money`, so the the `displayAmount` is not available. the displayAmount is needed for ticket: https://github.com/reactioncommerce/reaction-next-starterkit/pull/383 otherwise no currency symbol would be displayed.

## Solution

- Update schema, make `compareAtPrice` of type `Money`
- Add a resolver to for `compareAtPrice` fields on catalog products
- Add `compareAtPrice` to the `xformCurrencyExchangePricing` transform

## Breaking changes

- `compareAtPrice` is now an object of type `Money`.

## Testing
1. With "the" sample data loaded into reaction, and Open Exhange Rates configured
2. Run the following query

```graphql
{
  LightBulbShirtProduct: catalogItemProduct(slugOrId: "light-bulb-logo-t-shirt") {
    product {
      variants {
        title
        pricing {
          currencyExchangePricing(currencyCode: "EUR") {
            compareAtPrice {
              displayAmount
              amount
            }
            displayPrice
            price
          }
          price
          displayPrice
          compareAtPrice {
            displayAmount
            amount
          }
        }
        options {
          pricing {
            displayPrice
            compareAtPrice {
              displayAmount
              amount
            }
          }
        }
      }
    }
  }
}
```

3. You should get the following result. **You're looking for the `compareAtPrice` objects**

```json
{
  "data": {
    "LightBulbShirtProduct": {
      "product": {
        "variants": [
          {
            "title": "Light Bulb Logo T-Shirt",
            "pricing": [
              {
                "currencyExchangePricing": {
                  "compareAtPrice": {
                    "displayAmount": "24,21 €",
                    "amount": 24.21
                  },
                  "displayPrice": "21,62 €",
                  "price": 21.62
                },
                "price": 24.99,
                "displayPrice": "$24.99",
                "compareAtPrice": {
                  "displayAmount": "$27.99",
                  "amount": 27.99
                }
              }
            ],
            "options": [
              {
                "pricing": [
                  {
                    "displayPrice": "$24.99",
                    "compareAtPrice": {
                      "displayAmount": "$27.99",
                      "amount": 27.99
                    }
                  }
                ]
              },
              {
                "pricing": [
                  {
                    "displayPrice": "$24.99",
                    "compareAtPrice": {
                      "displayAmount": "$27.99",
                      "amount": 27.99
                    }
                  }
                ]
              },
              {
                "pricing": [
                  {
                    "displayPrice": "$24.99",
                    "compareAtPrice": {
                      "displayAmount": "$27.99",
                      "amount": 27.99
                    }
                  }
                ]
              }
            ]
          }
        ]
      }
    },
  }
}
```

### CODE REVIEW

**Comment by mikemurray** [comment]
@aldeed You can check the code
@kieckhafer This ticket should tested be merged before testing: https://github.com/reactioncommerce/reaction-next-starterkit/pull/383

File: imports/plugins/core/catalog/server/no-meteor/resolvers/ProductPricingInfo.js [hunk 1]
```diff
@@ -2,5 +2,6 @@ import { getXformedCurrencyByCode, xformCurrencyExchangePricing } from "@reactio
 
 export default {
   currency: (node) => getXformedCurrencyByCode(node.currencyCode),
-  currencyExchangePricing: (node, { currencyCode }, context) => xformCurrencyExchangePricing(node, currencyCode, context)
+  currencyExchangePricing: (node, { currencyCode }, context) => xformCurrencyExchangePricing(node, currencyCode, context),
+  compareAtPrice: (node) => ({ amount: node.compareAtPrice || 0, currencyCode: node.currencyCode })
```
[the remark this answers is not in the document]
Comment (mikemurray) [reply imports/plugins/core/catalog/server/no-meteor/resolvers/ProductPricingInfo.js]: Returning `null` results in the following error being logged.

```
reaction_1            | 18:07:02.972Z ERROR Reaction:  (errorId=cjnam23p800005as1922g2z3y, userId=null)
reaction_1            |   path: [
reaction_1            |     "catalogItemProduct",
reaction_1            |     "product",
reaction_1            |     "variants",
reaction_1            |     0,
reaction_1            |     "pricing",
reaction_1            |     0,
reaction_1            |     "compareAtPrice",
reaction_1            |     "amount"
reaction_1            |   ]
reaction_1            | 18:07:02.973Z ERROR Reaction:  (errorId=cjnam23p800015as1kr9zazd8, userId=null)
reaction_1            |   path: [
reaction_1            |     "catalogItemProduct",
reaction_1            |     "product",
reaction_1            |     "pricing",
reaction_1            |     0,
reaction_1            |     "compareAtPrice",
reaction_1            |     "amount"
reaction_1            |   ]
reaction_1            | Error: Cannot return null for non-nullable field Money.amount.
reaction_1            |     at completeValue (/opt/reaction/src/node_modules/graphql/execution/execute.js:638:13)
reaction_1            |     at completeValueWithLocatedError (/opt/reaction/src/node_modules/graphql/execution/execute.js:585:21)
reaction_1            |     at completeValueCatchingError (/opt/reaction/src/node_modules/graphql/execution/execute.js:554:12)
reaction_1            |     at resolveField (/opt/reaction/src/node_modules/graphql/execution/execute.js:500:10)
reaction_1            |     at /opt/reaction/src/node_modules/graphql/execution/execute.js:367:18
reaction_1            |     at Array.reduce (<anonymous>)
reaction_1            |     at executeFields (/opt/reaction/src/node_modules/graphql/execution/execute.js:364:42)
reaction_1            |     at collectAndExecuteSubfields (/opt/reaction/src/node_modules/graphql/execution/execute.js:781:10)
reaction_1            |     at completeObjectValue (/opt/reaction/src/node_modules/graphql/execution/execute.js:771:10)
reaction_1            |     at completeValue (/opt/reaction/src/node_modules/graphql/execution/execute.js:667:12)
reaction_1            |     at completeValueWithLocatedError (/opt/reaction/src/node_modules/graphql/execution/execute.js:585:21)
reaction_1            |     at completeValueCatchingError (/opt/reaction/src/node_modules/graphql/execution/execute.js:560:21)
reaction_1            |     at resolveField (/opt/reaction/src/node_modules/graphql/execution/execute.js:500:10)
reaction_1            |     at /opt/reaction/src/node_modules/graphql/execution/execute.js:367:18
reaction_1            |     at Array.reduce (<anonymous>)
reaction_1            |     at executeFields (/opt/reaction/src/node_modules/graphql/execution/execute.js:364:42)
```

[the remark this answers is not in the document]
Comment (aldeed) [reply imports/plugins/core/catalog/server/no-meteor/resolvers/ProductPricingInfo.js]: @mikemurray Return `null` for the whole resolver, not just the `amount`

**Comment by mikemurray** [comment]
@aldeed ready for a re-review

**Pull request merged**